docs/advanced_topics_performance_tuning.rst

```diff
@@ -83,6 +83,37 @@ For convenience, the DALI_BUFFER_GROWTH_FACTOR environment variable and the
 `nvidia.dali.backend.SetBufferGrowthFactor` Python function can be used to set the same
 growth factor for the host and the GPU buffers.
 
+
+Allocator Configuration
+-----------------------
+
+DALI uses several types of memory resources for various kinds of memory.
+
+For regular host memory, DALI uses (aligned) ``malloc`` for small allocations and a custom memory
+pool for large allocations (to prevent costly calls to ``mmap`` by ``malloc``).
+The maximum size of a host allocation that is allocated directly with ``malloc`` can be customized
+by setting ``DALI_MALLOC_POOL_THRESHOLD`` environment variable. If not specified, the value is
+either derived from environment variables controlling ``malloc`` or, if not found, a default value
+of 32M is used.
+
+For host pinned memory, DALI uses a stream-aware memory pool on top of ``cudaMallocHost``.
```

Comment:
Maybe you can link `https://developer.nvidia.com/blog/using-cuda-stream-ordered-memory-allocator-part-1/` for the reference.